pkg/aws/cloud.go

```diff
@@ -114,9 +108,41 @@ func NewCloud(cfg CloudConfig, metricsRegisterer prometheus.Registerer) (Cloud,
 		metricsCollector.InjectHandlers(&sess.Handlers)
 	}
 
+	ec2Service := services.NewEC2(sess)
+
+	if len(cfg.VpcID) == 0 {
+		nodeName := os.Getenv("NODENAME")
+		if strings.HasPrefix(nodeName, "i-") {
+			output, err := ec2Service.DescribeInstances(&ec2.DescribeInstancesInput{
+				InstanceIds: []*string{&nodeName},
+			})
+			if err != nil {
```

Comment:
we should change our logic here, if we don't have permission to describe the instance, we still should be able to fallback to metadata path. 
similarly, if we check for metadata first then instance id(see my next comment), we should change the logic to not fall when metadata is not available

Reply:
Also done.
Means less meaningful errors returned, but perhaps not very important anyway.